### DISCUSSION TITLE
One problem when I log in with icepy

### DISCUSSION THREAD

**Discussion by allicyn1221** [icesat2py/icepyx#429] [Q&A] [answered]
Hi, in April  I downloads data by icepy, things went very well. But today when I tried to log in to NASA i, it gave me this:

**Earthdata Login password: ········
---------------------------------------------------------------------------
JSONDecodeError                           Traceback (most recent call last)
Input In [33], in <cell line: 1>()
----> 1 region_a.earthdata_login('Allicynguo1221','guoyixin@sjtu.edu.cn')

File ~\anaconda3\envs\nsidc-tutorials\lib\site-packages\icepyx\core\query.py:851, in Query.earthdata_login(self, uid, email, s3token)
    846     assert (
    847         is_ec2() == True
    848     ), "You must be working from a valid AWS instance to use s3 data access"
    849     capability_url = "https://data.nsidc.earthdatacloud.nasa.gov/s3credentials"
--> 851 self._session = Earthdata(uid, email, capability_url).login()
    853 # DevNote: might make sense to do this part elsewhere in the future, but wanted to get it implemented for now
    854 if s3token == True:

File ~\anaconda3\envs\nsidc-tutorials\lib\site-packages\icepyx\core\Earthdata.py:140, in Earthdata.login(self, attempts)
    138 for i in range(attempts):
    139     try:
--> 140         self._start_session()
    141         break
    142     except KeyError:

File ~\anaconda3\envs\nsidc-tutorials\lib\site-packages\icepyx\core\Earthdata.py:78, in Earthdata._start_session(self)
     76 # check for a valid login
     77 try:
---> 78     json.loads(response.content)["token"]
     79 except KeyError:
     80     try:

File ~\anaconda3\envs\nsidc-tutorials\lib\json\__init__.py:346, in loads(s, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kw)
    341     s = s.decode(detect_encoding(s), 'surrogatepass')
    343 if (cls is None and object_hook is None and
    344         parse_int is None and parse_float is None and
    345         parse_constant is None and object_pairs_hook is None and not kw):
--> 346     return _default_decoder.decode(s)
    347 if cls is None:
    348     cls = JSONDecoder

File ~\anaconda3\envs\nsidc-tutorials\lib\json\decoder.py:337, in JSONDecoder.decode(self, s, _w)
    332 def decode(self, s, _w=WHITESPACE.match):
    333     """Return the Python representation of ``s`` (a ``str`` instance
    334     containing a JSON document).
    335 
    336     """
--> 337     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
    338     end = _w(s, end).end()
    339     if end != len(s):

File ~\anaconda3\envs\nsidc-tutorials\lib\json\decoder.py:355, in JSONDecoder.raw_decode(self, s, idx)
    353     obj, end = self.scan_once(s, idx)
    354 except StopIteration as err:
--> 355     raise JSONDecodeError("Expecting value", s, err.value) from None
    356 return obj, end

JSONDecodeError: Expecting value: line 1 column 1 (char 0)**

I would be very grateful if you could give me some help.

**Comment by JessicaS11** [comment, marked as answer]
Hello @allicyn1221  - could you please try upgrading to the most recent version of icepyx (v0.7.0; you can check which version you are running using `ipx.__version__`). There was a change on the NSIDC side regarding authentication (token types) that is handled in the latest release. I'm going to close this issue, but please feel free to reopen it if updating versions does not address your problem.

**Reply by allicyn1221** [reply]
Thanks for your answer. I just reinstalled the entire environment and updated the latest icepyx(0.7.0). But there's a new problem,
when I ran import `import icepyx as ipx`,  I got `ImportError: Can't determine version for numexpr`
Have you ever seen it before?

**Reply by allicyn1221** [reply]
I googled this situation and reinstalled the environment, installed packages(numexpr, bottleneck). It didn't work at all. I also tried reinstalled ipx, it didn't work either.

**Reply by JessicaS11** [reply]
Could you please provide the lines of code or process you are using to create your environment? I'm not able to recreate your issue by creating a new environment, which may be the result of what tools we're using to set up our environment.
